### PR TITLE
[5.5] Improvements: comparison, phpdoc, fixed small inaccuracies

### PR DESCRIPTION
**Pull request by dmirogin**
Improvements: comparison, phpdoc, fixed small inaccuracies

### CODE REVIEW

**Comment by 36864** [comment]
I suggest splitting this into smaller PRs that each fix one type of thing.
One for fixing the phpdocs, one for the strict comparison checks and one for each of the other fixes explaining exactly what you're fixing.

File: src/Illuminate/Database/Query/Builder.php [hunk 1]
```diff
@@ -1228,7 +1228,7 @@ public function addWhereExistsQuery(self $query, $boolean = 'and', $not = false)
     {
         $type = $not ? 'NotExists' : 'Exists';
 
-        $this->wheres[] = compact('type', 'operator', 'query', 'boolean');
```
Comment (browner12) [inline src/Illuminate/Database/Query/Builder.php]: why are you removing this?

Comment (dmirogin) [reply src/Illuminate/Database/Query/Builder.php]: this function doesn't contain variable with name `operator`

**Comment by browner12** [comment]
agreed. this is too wide in scope.

also, based on past behavior, it is very unlikely for your strict comparison changes to be merged.

File: src/Illuminate/Database/Eloquent/Factory.php [hunk 2]
```diff
@@ -233,7 +233,7 @@ public function offsetGet($offset)
      *
      * @param  string  $offset
      * @param  callable  $value
-     * @return void
+     * @return $this
```
Comment (sisve) [inline src/Illuminate/Database/Eloquent/Factory.php]: This is wrong. ArrayAccess::offsetSet returns void. https://secure.php.net/manual/en/arrayaccess.offsetset.php

File: src/Illuminate/Filesystem/FilesystemAdapter.php [hunk 3]
```diff
@@ -240,7 +240,7 @@ public function getVisibility($path)
      *
      * @param  string  $path
      * @param  string  $visibility
-     * @return void
+     * @return bool
```
Comment (sisve) [inline src/Illuminate/Filesystem/FilesystemAdapter.php]: This is wrong. The contract states that this should return void. https://github.com/laravel/framework/blob/bd352a0d2ca93775fce8ef02365b03fc4fb8cbb0/src/Illuminate/Contracts/Filesystem/Filesystem.php#L64

**Comment by sisve** [comment]
The fact that you've missed these obvious cases and quickly commit a reversal of the pointed out problems indicate that you've just run a tool that suggested changes without thinking this through.

At least target the master branch so we don't have to debug and document any breaking changes in the 5.5 branch. I'm pretty sure there are several in this pr. Here's an example on how strict comparisons may break stuff: https://3v4l.org/RQaO4

**Comment by dmirogin** [comment]
Yes, you are right. I thoroughly went by those problems, but did not notice this problems. 
Thanks for pointing on this comparison cause. 
Sorry for wasting your time.

**Pull request closed without merging**